Step1 구현

kineo2k opened the pull request:
Step1 구현했습니다.
리뷰 부탁드려요 ^^

kineo2k committed:
Step1 구현

javajigi commented on src/main/java/com/mrblue/codesquad/javaracingcar/Car.java:
```diff
@@ -0,0 +1,53 @@
+ * 자동차 이동 알고리즘 관리
+ */
+// TODO : 테스트 방법을 몰라서 Private 메소드도 전부 public으로 설정
+public class Car {
```
모든 메소드를 public으로 두지 말고 최소한의 범위만 허용하도록 한다.

kineo2k replied:
시작이 늦었습니다 ^^;
가능한 모든 인터페이스를 private으로  수정했습니다.

javajigi commented on src/main/java/com/mrblue/codesquad/javaracingcar/Car.java:
```diff
@@ -0,0 +1,53 @@
+				 .mapToObj(idx -> this.randomNumber())
+				 .filter(this::isAllowRange)
+				 .reduce((first, second) -> first + second)
+				 .ifPresent(this::printResult);
```
로직 구현하는 곳에서 View를 바로 호출하지 말고 둘을 분리한다.
이와 같이 구현할 경우 테스트하기 어려움.

kineo2k replied:
로직과 뷰 호출부를 분리했습니다.

javajigi commented on src/main/java/com/mrblue/codesquad/javaracingcar/Car.java:
```diff
@@ -0,0 +1,53 @@
+				.append("입니다.")
+				.toString();
+
+		System.out.println(resultMessage);
```
View 로직을 별도의 클래스로 분리한다.

kineo2k replied:
뷰로직을 별도 클래스로 분리했습니다.
간단한 출력 구문이라서 뷰라는 인지를 못하고 그냥 코딩 했네요.
많이 배웁니다 ^^

javajigi reviewed (changes requested):
구현 잘 하셨네요. 연습 삼아 피드백 몇개 남겼어요.

kineo2k committed:
- step1 리팩토링
- RacingGame, Car 클래스에서 외부에 공개할 필요가 없는 인터페이스는 private 처리
- private 메서드를 테스트하기 위하여 com.mrblue.codesquad.testutil.TestUtil 클래스 추가
- 로직과 뷰를 분리하기 위해서 com.mrblue.codesquad.racingcar.ConsoleIO 클래스 추가
- Car 클래스 move 메서드에서 로직과 뷰 호출을 분리
